xbmc/video/VideoDatabase.cpp

```diff
@@ -4647,12 +4655,13 @@ void CVideoDatabase::GetSameVideoItems(const CFileItem& item, CFileItemList& ite
     // note 2: for type 'tmdb' the same value may be used for a movie and a tv episode, only
     // distinguished by media_type.
     // @todo make the (value,type) pairs truly unique
-    m_pDS->query(PrepareSQL("SELECT DISTINCT media_id "
-                            "FROM uniqueid "
-                            "WHERE (media_type, value, type) IN "
-                            "  (SELECT media_type, value, type "
-                            "  FROM uniqueid WHERE media_id = %i AND media_type = '%s') ",
-                            dbId, mediaType.c_str()));
+    m_pDS->query(
+        PrepareSQL("SELECT DISTINCT media_id "
+                   "FROM uniqueid "
+                   "WHERE (media_type, value, type) IN "
+                   "  (SELECT media_type, value, type "
+                   "  FROM uniqueid WHERE media_id = %i AND media_type = '%s' AND value != '') ",
+                   dbId, mediaType.c_str()));
```

Comment:
If I remember correctly that query caused a lengthy discussion shortly after videoversions merge, and I will need to look closer.
Until then it's safer to validate the inputs from scraper and nfo inputs so the case doesn't happen in the first place, and to remove the existing cases from the db (db bump needed)

Reply:
That was not an approval either. It was a "wait a bit". Don't be surprised if I make another change on top because as I wrote twice it's a tricky query. I also don't know what kind of testing you did.

Reply:
> I also don't know what kind of testing you did.

> **How has this been tested?**
> Tested that existing movies with empty IDs no longer appears as suggest from Manage Versions / Add Version dialog.
> 
> Before, selecting any movie with an empty ID was obtained a list of all other movies with empty ID's as suggestion for "Add Version".
> 
> Tested also that movies with same unique ID continues matching as suggestion for "Add Version" from "Manage Versions".

Reply:
That's not what I meant, my concern is about non-trivial cases. As I wrote, I remember some 'trickyness'.